src/code/z_actor.c

```diff
@@ -2226,14 +2258,14 @@ void Actor_DeactivateLens(PlayState* play) {
     }
 }
 
-void func_800B9120(ActorContext* actorCtx) {
-    s32 phi_v0 = CURRENT_DAY * 2;
+void Actor_InitHalfDaysBit(ActorContext* actorCtx) {
+    s32 halfDayCount = CURRENT_DAY * 2;
 
     if (gSaveContext.save.time < CLOCK_TIME(6, 0) || gSaveContext.save.time > CLOCK_TIME(18, 0)) {
-        phi_v0++;
+        halfDayCount++;
     }
 
-    actorCtx->unkC = 0x200 >> phi_v0;
+    actorCtx->halfDaysBit = 0x200 >> halfDayCount;
```

Comment:
```suggestion
    actorCtx->halfDaysBit = HALFDAYBIT_DAY0_DAWN >> halfDayCount;
```
I wonder if this is appropriate. If you add more days, you'd want the first bit to be the max value, which in this case is always `HALFDAYBIT_DAY0_DAWN`